### ISSUE TITLE
Timed out statements raises poor exception content

### ISSUE THREAD

**Issue by gmiejski** [aio-libs/aiopg#537]
Versions:

```
aiopg==0.16.0
psycopg2==2.7.7
```

When statement gets cancelled after setting a timeout, only a coroutine `CancelledError` is raised without any context. I would expect some more information why the query has been cancelled:

Reproduce:

```
import asyncio

from aiopg.sa import create_engine


def create_eng():
    engine = create_engine(
        "postgresql://postgres:postgres@localhost:5432/postgres?sslmode=disable",
        options="-c statement_timeout={}".format(1000),
    )
    loop = asyncio.get_event_loop()
    engine = loop.run_until_complete(engine)
    return engine


async def main(engine):
    async with engine.acquire() as conn:
        async with conn.execute("SELECT * FROM pg_sleep(5)") as result:
            rows = await result.fetchall()
            print(rows)


asyncio.get_event_loop().run_until_complete((main(create_eng())))
```

Expected output:
```
any message saying that execute statement timed out
```

Actual output:

```
Traceback (most recent call last):
  File "aiopg_timeout_minimal_example.py", line 23, in <module>
      asyncio.get_event_loop().run_until_complete((main(create_eng())))
  File "/lib/python3.7/asyncio/base_events.py", line 573, in run_until_complete
      return future.result()
concurrent.futures._base.CancelledError
```

**Comment by vir-mir** [comment]
@gmiejski Hi.
unfortunately `CanceledError` does not support extended traceback. You can read it [here](https://github.com/python/cpython/blob/master/Lib/asyncio/base_events.py#L582) and [here](https://github.com/python/cpython/blob/master/Lib/asyncio/futures.py#L178).

But I think. We can add `loop.call_exception_handler` which can be overridden with `loop.set_exception_handler` and with the help of the standard python logger,  you can send errors in sentry... [the here](https://github.com/aio-libs/aiopg/blob/ef1cabeeb7023841184f23197298215d1d45209d/aiopg/connection.py#L243)

**Comment by Pliner** [comment]
Resolved by #537
